src/identity/identity.ts

```diff
@@ -227,14 +227,30 @@ export class Identity extends Logger {
 
     // apply the identity as key to the keyHolder contract
     const keyHolderContract = await contractLoader.loadContract('KeyHolder', activeIdentity);
-    await executor.executeContractTransaction(
-      keyHolderContract,
-      'addKey',
-      { from: underlyingAccount },
-      nameResolver.soliditySha3(accountId),
-      1,
-      1,
-    );
+    const version = await this.options.executor.executeContractCall(keyHolderContract, 'VERSION_ID');
+    if (version === null) {
+      // older identities only use purose 1 for granting access
+      await executor.executeContractTransaction(
+        keyHolderContract,
+        'addKey',
+        { from: underlyingAccount },
+        nameResolver.soliditySha3(accountId),
+        1,
+        1,
+      );
+    } else if (version.eq('1')) {
+      // new identities have split permissions into different purposes
+      await executor.executeContractTransaction(
+        keyHolderContract,
+        'addMultiPurposeKey',
+        { from: underlyingAccount },
+        nameResolver.soliditySha3(accountId),
+        [1, 2],
```

Comment:
reading the code it's absolutely unclear, what purpose 1, 2 or 3 is representing.
I would suggest to put it in a constant or enum mapping and use then semantic variables accordingly.
`enum Purpose { Owner, Admin, Viewer }` ...